discovery-infra/test_infra/helper_classes/kube_helpers/agent.py

```diff
@@ -114,3 +114,57 @@ def _attempt_to_get_status() -> dict:
     def approve(self) -> None:
         self.patch(approved=True)
         logger.info("approved agent %s", self.ref)
+
+    @staticmethod
+    def wait_for_agents_to_install(agents, nodes_number, timeout: Union[int, float] = DEFAULT_WAIT_FOR_CRD_STATUS_TIMEOUT) -> None:
+        Agent.wait_till_all_agents_are_in_status(
+            agents=agents,
+            nodes_count=nodes_number,
+            status=consts.AgentStatus.VALIDATED,
+            timeout=timeout,
+        )
+        Agent.wait_till_all_agents_are_in_status(
+            agents=agents,
+            nodes_count=nodes_number,
+            status=consts.AgentStatus.READY_FOR_INSTALLATION,
+            timeout=timeout,
+        )
+        Agent.wait_till_all_agents_are_in_status(
+            agents=agents,
+            nodes_count=nodes_number,
+            status=consts.AgentStatus.INSTALLED,
+            timeout=timeout,
+        )
+
+    @staticmethod
+    def are_agents_in_status(
+                    agents, nodes_count, status,
+    ) -> bool:
+        agents_in_status = [agent for agent in agents for condition in  agent.status()["conditions"] if condition["type"] == status]
```

Comment:
Move the logger over here, so we will always now what happens.

Reply:
Fixed. I had to also check for the status along with the type in the agent condition.

```
2021-06-23 15:58:32,932 INFO       - 140351335046016 - Asked agents to have the status [('ReadyForInstallation', 'True')] and currently agent statuses are [('SpecSynced', 'True'), ('Connected', 'True'), ('ReadyForInstallation', 'False'), ('Validated', 'True'), ('Installed', 'False')] 	(/home/test/assisted-test-infra/discovery-infra/test_infra/helper_classes/kube_helpers/agent.py:146)
2021-06-23 15:58:37,937 INFO       - 140351335046016 - Asked agents to have the status [('ReadyForInstallation', 'True')] and currently agent statuses are [('SpecSynced', 'True'), ('Connected', 'True'), ('ReadyForInstallation', 'True'), ('Validated', 'True'), ('Installed', 'False')] 	(/home/test/assisted-test-infra/discovery-infra/test_infra/helper_classes/kube_helpers/agent.py:146)
2021-06-23 15:58:37,940 INFO       - 140351335046016 - Now Wait till agents have status as Installed 	(/home/test/assisted-test-infra/discovery-infra/test_infra/helper_classes/kube_helpers/agent.py:162)
2021-06-23 15:58:37,942 INFO       - 140351335046016 - Asked agents to have the status [('Installed', 'True')] and currently agent statuses are [('SpecSynced', 'True'), ('Connected', 'True'), ('ReadyForInstallation', 'True'), ('Validated', 'True'), ('Installed', 'False')] 	(/home/test/assisted-test-infra/discovery-infra/test_infra/helper_classes/kube_helpers/agent.py:146)
2021-06-23 15:58:42,947 INFO       - 140351335046016 - Asked agents to have the status [('Installed', 'True')] and currently agent statuses are [('SpecSynced', 'True'), ('Connected', 'True'), ('ReadyForInstallation', 'True'), ('Validated', 'True'), ('Installed', 'False')] 	(/home/test/assisted-test-infra/discovery-infra/test_infra/helper_classes/kube_helpers/agent.py:146)
2021-06-23 15:58:47,952 INFO       - 140351335046016 - Asked agents to have the status [('Installed', 'True')] and currently agent statuses are [('SpecSynced', 'True'), ('Connected', 'True'), ('ReadyForInstallation', 'True'), ('Validated', 'True'), ('Installed', 'False')] 	(/home/test/assisted-test-infra/discovery-infra/test_infra/helper_classes/kube_helpers/agent.py:146)
2021-06-23 15:58:52,957 INFO       - 140351335046016 - Asked agents to have the status [('Installed', 'True')] and currently agent statuses are [('SpecSynced', 'True'), ('Connected', 'True'), ('ReadyForInstallation', 'False'), ('Validated', 'True'), ('Installed', 'False')] 	(/home/test/assisted-test-infra/discovery-infra/test_infra/helper_classes/kube_helpers/agent.py:146)
2021-06-23 15:58:57,962 INFO       - 140351335046016 - Asked agents to have the status [('Installed', 'True')] and currently agent statuses are [('SpecSynced', 'True'), ('Connected', 'True'), ('ReadyForInstallation', 'False'), ('Validated', 'True'), ('Installed', 'False')] 	(/home/test/assisted-test-infra/discovery-infra/test_infra/helper_classes/kube_helpers/agent.py:146)
2021-06-23 15:59:02,967 INFO       - 140351335046016 - Asked agents to have the status [('Installed', 'True')] and currently agent statuses are [('SpecSynced', 'True'), ('Connected', 'True'), ('ReadyForInstallation', 'False'), ('Validated', 'True'), ('Installed', 'False')] 	(/home/test/assisted-test-infra/discovery-infra/test_infra/helper_classes/kube_helpers/agent.py:146)
2021-06-23 15:59:07,972 INFO       - 140351335046016 - Asked agents to have the status [('Installed', 'True')] and currently agent statuses are [('SpecSynced', 'True'), ('Connected', 'True'), ('ReadyForInstallation', 'False'), ('Validated', 'True'), ('Installed', 'False')] 	(/home/test/assisted-test-infra/discovery-infra/test_infra/helper_classes/kube_helpers/agent.py:146)
2021-06-23 15:59:12,978 INFO       - 140351335046016 - Asked agents to have the status [('Installed', 'True')] and currently agent statuses are [('SpecSynced', 'True'), ('Connected', 'True'), ('ReadyForInstallation', 'False'), ('Validated', 'True'), ('Installed', 'False')] 	(/home/test/assisted-test-infra/discovery-infra/test_infra/helper_classes/kube_helpers/agent.py:146)

```